lib/mongoid/attributes.rb

```diff
@@ -353,30 +352,6 @@ def unalias_attribute(name)
 
     private
 
-    # Validates an attribute value as being assignable to the specified field.
-    #
-    # For now, only Hash and Array fields are validated, and the value is
-    # being checked to be of an appropriate type (i.e. either Hash or Array,
-    # respectively, or nil).
-    #
-    # This method takes the name of the field as stored in the document
-    # in the database, not (necessarily) the Ruby method name used to read/write
-    # the said field.
-    #
-    # @param [ String, Symbol ] field_name The name of the field.
-    # @param [ Object ] value The value to be validated.
-    def validate_attribute_value(field_name, value)
```

Comment:
_(Other thread was closed)_ This is a drive-by-change in this PR:
- MONGOID-5222 references `mongoize` method specifically.
- This method `validate_attribute_value` does not call `mongoize` (and vice-versa.)

Suggest to revisit changes here as part of [MONGOID-5382](https://jira.mongodb.org/browse/MONGOID-5382). This PR is large already and there's no reason to rush this change in.

Reply:
@p-mongo My comment here is politely-worded, on-topic, and is a valid concern. I'd appreciate if the MongoDB team can leave it unminimized and write a brief response when time allows. Thank you!

Reply:
@johnnyshields The reasoning for this change is as follows:

1. The overall epic being worked on is consistent type conversion of values (https://jira.mongodb.org/browse/MONGOID-5332).
2. For Mongoid 8.0, we decided to replace unconvertable input with nil for the value written into `attributes`. The original input is stored in `attributes_before_type_cast`.
3. This behavior applies to all field types, per the goal of the epic.
4. This PR implements the nil replacement behavior as just outlined.
5. I believe there is a valid use case for erroring (i.e. raising an exception) during type conversion instead of replacing with nil (https://jira.mongodb.org/browse/MONGOID-5382).
6. Erroring is not part of this PR and is not technically required for 8.0, since it can potentially be implemented as a compatible change in 8.1.

Your argument that the raise for non-Array/Hash incoming values for Array/Hash fields is out of scope of this PR because it's done in a method not called `mongoize` is nitpicking. Your opinion on this matter is noted, and your repeated posting of this opinion is off topic or if you prefer, spam.

If you wish to propose an adjustment to assignment behavior as explained in https://jira.mongodb.org/browse/MONGOID-5222, please leave a comment in that ticket. You need to write a *constructive* comment, meaning it needs to address the requirements in a plausibly better way than the existing implementation being proposed in this PR.

If you have any thoughts you haven't already expressed in regard to failing attribute assignment and/or attribute reads from database with exceptions, please add those to https://jira.mongodb.org/browse/MONGOID-5382, also in a constructive manner. Your current position there is that that ticket is not needed at all. Your opinion that instead of having an option for any field type to fail assignments of unconvertable input Mongoid should retain ad-hoc checks for some types scattered through the codebase is noted, we acknowledge that this is existing behavior of Mongoid and we are looking to improve it.